tests/bluetooth/bsim/nrf_auraconfig/tester/src/nrf_auraconfig_test.c

```diff
@@ -0,0 +1,1164 @@
+
+				if (chan_allocation & bit_val) {
+					LOG_DBG("\t\t\tBIS %d: %s", info->bis_index,
+						bt_audio_location_bit_to_str(bit_val));
+				}
+			}
+		}
+
+		info->bis_index += 1;
+
+	} else {
+		return false;
+	}
+
+	return true;
+}
+
+static bool print_base_subgroup_cb(const struct bt_bap_base_subgroup *subgroup, void *user_data)
+{
+	int ret;
+	struct bt_bap_base_codec_id codec_id;
+	struct bt_audio_codec_cfg codec_cfg;
+	uint8_t *sub_index = user_data;
+
+	LOG_DBG("Subgroup: %d", *sub_index);
+
+	ret = bt_bap_base_get_subgroup_codec_id(subgroup, &codec_id);
+	if (ret < 0) {
+		return false;
+	}
+
+	struct stream_info info = {
+		.codec_id = codec_id,
+		.sub_index = *sub_index,
+		.bis_index = 0,
+	};
+
+	ret = bt_bap_base_subgroup_codec_to_codec_cfg(subgroup, &codec_cfg);
+	if (ret == 0) {
+		LOG_DBG("\tCodec specific configuration:");
+
+		if (codec_cfg.data_len == 0U) {
+			LOG_DBG("\t\tNone");
+		} else if (codec_cfg.id == BT_HCI_CODING_FORMAT_LC3) {
+			int ret;
+
+			ret = bt_audio_codec_cfg_get_freq(&codec_cfg);
+			if (ret >= 0) {
+				dut.subgroups[info.sub_index].sampling_rate_hz =
+					bt_audio_codec_cfg_freq_to_freq_hz(ret);
+				LOG_DBG("\t\tSampling rate: %u Hz",
+					dut.subgroups[info.sub_index].sampling_rate_hz);
+			}
+
+			ret = bt_audio_codec_cfg_get_frame_dur(&codec_cfg);
+			if (ret >= 0) {
+				dut.subgroups[info.sub_index].frame_duration_us =
+					bt_audio_codec_cfg_frame_dur_to_frame_dur_us(ret);
+				LOG_DBG("\t\tFrame duration: %u us",
+					dut.subgroups[info.sub_index].frame_duration_us);
+			}
+
+			ret = bt_audio_codec_cfg_get_octets_per_frame(&codec_cfg);
+			if (ret >= 0) {
+				dut.subgroups[info.sub_index].octets_per_frame = (uint16_t)ret;
+				LOG_DBG("\t\tOctets per frame: %u",
+					dut.subgroups[info.sub_index].octets_per_frame);
+			}
+
+			ret = bt_bap_base_get_subgroup_bis_count(subgroup);
+			if (ret >= 0) {
+				dut.subgroups[info.sub_index].num_bis = (uint8_t)ret;
+				LOG_DBG("\t\tNumber of BISes: %u",
+					dut.subgroups[info.sub_index].num_bis);
+			}
+
+			LOG_DBG("\t\tLocation:");
+			ret = bt_bap_base_subgroup_foreach_bis(subgroup, print_base_subgroup_bis_cb,
+							       &info);
+			if (ret < 0) {
+				return false;
+			}
+
+		} else {
+			LOG_DBG("\t\tUnsupported codec");
+		}
+
+		LOG_DBG("\tCodec specific metadata:");
+
+		if (codec_cfg.meta_len == 0U) {
+			LOG_DBG("\tNone");
+		} else {
+			const uint8_t *data;
+
+			ret = bt_audio_codec_cfg_meta_get_stream_context(&codec_cfg);
+			if (ret >= 0) {
+				dut.subgroups[info.sub_index].context = ret;
+				for (size_t i = 0U; i < 16; i++) {
+					const uint16_t bit_val = BIT(i);
+
+					if (ret & bit_val) {
+						LOG_DBG("\t\tContext: %s",
+							bt_audio_context_bit_to_str(bit_val));
+					}
+				}
+			}
+
+			ret = bt_audio_codec_cfg_meta_get_program_info(&codec_cfg, &data);
+			if (ret >= 0) {
+				char program_info[255] = {'\0'};
+
+				if (ret > sizeof(program_info)) {
+					ret = sizeof(program_info);
```

Comment:
This is to make sure we do not try to copy more data to program_info than what it can take. I will change the if to:
```C
if (ret > PROGRAM_INFO_MAX_LEN) {
        ret = PROGRAM_INFO_MAX_LEN - 1;
}
```
maybe it makes more sense then? Ret is used in the memcpy below